pkg/manager/internal.go

```diff
@@ -368,6 +370,18 @@ func (cm *controllerManager) GetWebhookServer() *webhook.Server {
 	return cm.webhookServer
 }
 
+func (cm *controllerManager) GetRESTClient(obj runtime.Object) (rest.Interface, error) {
```

Comment:
Nit: Should we suffix the method name with `For` so that it is more clear that this is for a specific object?

```suggestion
func (cm *controllerManager) GetRESTClientFor(obj runtime.Object) (rest.Interface, error) {
```

Reply:
Not to be pedantic but wouldn't this be `GetRESTClientForGVK()`?
The client is for the GVK, and not just the object.
Where the interface would be `RESTClientForGVK(gvk schema.GroupVersionKind)`. 

Slightly more verbose to actually use though a bit clearer imo:
```Go
gvk, err := apiutil.GVKForObject(obj, mgr.GetScheme())
...
rc, err := mgr.RESTClientForGVK(gvk)
```

Reply:
I don't disagree that it could be confusing, but good go doc could go a long way. There is also some precedent already for using a `runtime.Object` to mean all objects of this type. See [`Builder.For()`](https://github.com/kubernetes-sigs/controller-runtime/blob/53626115bd889e6c9039aaaad9b482a713de5bd5/pkg/builder/controller.go#L72-L84).